### PR TITLE
Add reversible encoding flag

### PR DESCRIPTION
**Pull request by simonjaeger**
Allowing for `UseReversibleEncoding` on encoders to be set with a flag `--re|--reversibleencoding`.

### CODE REVIEW

**Review by peterbryntesson — APPROVED** [review]
Mostly parameter passing LGTM

**Review by marcschier — APPROVED** [review]

**Comment by mregen** [comment]
Whats the use case?

**Comment by simonjaeger** [comment]
@mregen 

1. The ability to get the Locale of LocalizedText objects. A customer needs to know the language of error messages of the machines (English or German). For example, the Message property of an Event is only encoded with the Locale when UseReversibleEncoding is set.
1. To be able to use the JsonDecoderEx to process the output of the OPC Publisher, related to https://github.com/Azure/Industrial-IoT/issues/1005

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/IMessageEncoder.cs [hunk 1]
```diff
@@ -43,17 +43,19 @@ public interface IMessageEncoder {
         /// </summary>
         /// <param name="message"></param>
         /// <param name="maxMessageSize"></param>
+        /// <param name="useReversibleEncoding"></param>
```
Comment (koepalex) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/IMessageEncoder.cs:46]: Please don't follow bad practices, rather remove the <param /> comment elments than adding a new empty one

Comment (simonjaeger) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/IMessageEncoder.cs:46]: Agreed! I have fixed it. Removing comments causes build error. Filled out all params instead.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs [hunk 2]
```diff
@@ -100,7 +101,7 @@ public class NetworkMessageEncoder : IMessageEncoder {
                     var helperEncoder = new JsonEncoderEx(helperWriter, encodingContext) {
                         UseAdvancedEncoding = true,
                         UseUriEncoding = true,
-                        UseReversibleEncoding = false
+                        UseReversibleEncoding = useReversibleEncoding
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs:104]: should always be true

**Comment by mregen** [comment]
According to the standard, NetworkMessage is always reversible, only DataValue per configuration.. need to double check how this is implemented in main

**Comment by mregen** [comment]
we need to have a quick discussion on our side how to address this across all releases

**Comment by koepalex** [comment]
As discussed with @mregen 

* Current implementation is slightly wrong
  * NetworkMessage and DataSetMessage need to be reversible encoded but this will only affect `StatusCode` and StatusCode is omitted  if it is *OK*
  * `DataSetFieldContentMask` is set to _DataValue_ which enforces an *non-reversible*  encoding

Ideal solution would be to configure `DataSetfieldContentMask` as _Variant_  and the encoder should distinguish based on it but encode NetworkMessage and DataSetMessage always as reversible (independent from payload).

Suggested solution: to set behavior as global setting, seems to be ok, because DataSetFieldContentMask isn't configurable at the moment.

**Review by koepalex — APPROVED** [review]

**Review by mregen — APPROVED** [review]
🥇

**Pull request merged**